Content.Server/_Moffstation/ReadyManifest/ReadyManifestSystem.cs

```diff
@@ -0,0 +1,165 @@
+        }
+    }
+
+    private void UpdateByPlayer(NetUserId userId)
+    {
+        if (!_prefsManager.TryGetCachedPreferences(userId, out var preferences))
+        {
+            return;
+        }
+
+        var profile = (HumanoidCharacterProfile)preferences.SelectedCharacter;
+        var player = _playerManager.GetSessionById(userId);
+        var profileJobs = FilterPlayerJobs(profile, player);
+
+        var isReady = _gameTicker.PlayerGameStatuses[userId] == PlayerGameStatus.ReadyToPlay;
+
+        foreach (var job in profileJobs)
+        {
+            if (!_jobCounts.TryGetValue(job, out var counts) ||
+                !profile.JobPriorities.TryGetValue(job, out var priority))
+            {
+                continue;
+            }
+
+            if (priority is not (JobPriority.High or JobPriority.Medium or JobPriority.Low))
+                continue;
+
+            int delta = isReady ? 1 : -1;
+
+            _jobCounts[job] = priority switch
+            {
+                JobPriority.High => counts with { High = counts.High + delta },
+                JobPriority.Medium => counts with { Medium = counts.Medium + delta },
+                JobPriority.Low => counts with { Low = counts.Low + delta },
+                _ => counts
+            };
+        }
+    }
+
+    private List<ProtoId<JobPrototype>> FilterPlayerJobs(HumanoidCharacterProfile profile, ICommonSession player)
+    {
+        var jobs = profile.JobPriorities.Keys.Select(k => new ProtoId<JobPrototype>(k)).ToList();
+        List<ProtoId<JobPrototype>> priorityJobs = [];
+        foreach (var job in jobs)
+        {
+            var priority = profile.JobPriorities[job];
+            // For jobs that are rolled before others such as Command, we want to check for any priority since they'll always be filled
+            if ((priority == JobPriority.High ||
+                 _prototypeManager.Index(job).Weight > 0 &&
+                 priority > JobPriority.Never) &&
+                _playTimeTracking.IsAllowed(player, job))
+            {
+                priorityJobs.Add(job);
+            }
+        }
+        return priorityJobs;
+    }
+
+    public Dictionary<ProtoId<JobPrototype>, (int High, int Medium, int Low)> GetReadyManifest()
+    {
+        return _jobCounts;
+    }
+
+    public void OpenEui(ICommonSession session)
+    {
+        if (_openEuis.ContainsKey(session))
+        {
+            return;
+        }
+
+        var eui = new ReadyManifestEui(this);
+        _openEuis.Add(session, eui);
+        _euiManager.OpenEui(eui, session);
+        eui.StateDirty();
+    }
+
+    private void UpdateEuis()
+    {
+        foreach (var (_, eui) in _openEuis)
+        {
+            eui.StateDirty();
+        }
+    }
+
+    public void CloseEui(ICommonSession session)
+    {
+        if (!_openEuis.TryGetValue(session, out var eui))
+        {
+            return;
+        }
+
+        _openEuis.Remove(session);
+        eui.Close();
+    }
```

Comment:
```suggestion
    public void CloseEui(ICommonSession session)
    {
        if (!_openEuis.Remove(session, out var eui))
        {
            return;
        }

        eui.Close();
    }
```

`Remove` with a non-null key is infallible.